pkg/util/file_indexer.go

```diff
@@ -8,35 +8,59 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 const fileIndexDirectory = ".odo"
 const fileIndexName = "odo-file-index.json"
 
+// FileIndex holds the file index used for storing local file state change
+type FileIndex struct {
+	metav1.TypeMeta
+	FileMap map[string]FileData
+}
+
+// NewFileIndex returns a fileIndex
+func NewFileIndex() *FileIndex {
+
+	return &FileIndex{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "FileIndex",
+			APIVersion: "v1",
```

Comment:
@kadel Should we go with semantic version?

Reply:
I think that this is ok.
We can use the same versioning schema that Kuberentes uses - https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning 

> Stable level:
The version name is vX where X is an integer.